src/openapi/streaming/subscription-queue.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Queue (FIFO) of pending subscription actions. Allows for easier abstraction of managing list of pending actions.
+ * In addition to standard queue implementation, provides:
+ *  - merging consecutive duplicated actions
+ *  - merging of uneccessary combinations: ACTION_SUBSCRIBE action before ACTION_UNSUBSCRIBE, is merged to ACTION_UNSUBSCRIBE.
+ *
+ * @module saxo/openapi/streaming/subscription queue
+ * @ignore
+ */
+
+import {
+	ACTION_SUBSCRIBE,
+	ACTION_UNSUBSCRIBE
+} from './subscription-actions';
+
+/**
+ * Queue (FIFO) for storing pending subscription actions.
+ * @param maxSize {Number} - Maximum queue size. Defaults to 2, which works best for current needs.
+ * @constructor
+ */
+function SubscriptionQueue(maxSize = 2) {
+	this.items = [];
+	this.maxSize = maxSize;
+}
+
+/**
+ * Add pending action to a queue and making sure that we remove unecessary actions like:
+ * - duplicates.
+ * - ACTION_SUBSCRIBE action before ACTION_UNSUBSCRIBE.
+ *
+ * @param {Number} action - action to add to the queue.
+ */
+SubscriptionQueue.prototype.enqueue = function(action) {
+	this.items.push(action);
+
+	if (this.items.length < this.maxSize) {
+		return;
+	}
+
+	for (let i = this.items.length - 2; i >= 0; i--) {
+		let task = this.items[i];
+		let nextTask = this.items[i + 1];
+
+		if (task === ACTION_SUBSCRIBE && nextTask === ACTION_UNSUBSCRIBE || task === nextTask) {
+			this.items.splice(i, 1);
```

Comment:
It will. Do we need initial subscribe, when it's surpassed by unsubscribe? 
Test case:
https://github.com/michax/openapi-clientlib-js/blob/ff4ec8cd774920bf5da19bc02931eaabdc1d640c/test/unit/openapi/streaming/subscription-queue-spec.js#L37